openmc/settings.py

```diff
@@ -927,6 +928,16 @@ def _create_cutoff_subelement(self, root):
 
     def _create_entropy_mesh_subelement(self, root):
         if self.entropy_mesh is not None:
+            # use default heuristic for entropy mesh if not set by user
+            if self.entropy_mesh.dimension is None:
+                    if self.particles is None:
+                        raise RuntimeError("Number of particles must be set in order to " \
+                          "use entropy mesh dimension heuristic")
+                    else:
+                        n = ceil((self.particles / 20.0)**(1.0 / 3.0))
+                        d = len(self.entropy_mesh.lower_left)
+                        self.entropy_mesh.dimension = (n,)*d
```

Comment:
Looks like this is too indented
```suggestion
                if self.particles is None:
                    raise ValueError("Number of particles must be set in order to "
                                     "use entropy mesh dimension heuristic")
                else:
                    n = ceil((self.particles / 20.0)**(1.0 / 3.0))
                    d = len(self.entropy_mesh.lower_left)
                    self.entropy_mesh.dimension = (n,)*d
```